cluster-autoscaler/cloudprovider/ionoscloud/client.go

```diff
@@ -160,80 +137,87 @@ func (p customClientProvider) GetClient() (APIClient, error) {
 	if len(cloudConfig.Tokens) == 0 {
 		return nil, fmt.Errorf("missing tokens for cloud config %s", filepath.Base(files[0]))
 	}
-	return NewAPIClient(cloudConfig.Tokens[0], p.endpoint, p.userAgent, p.insecure), nil
+	cfg := *c.cfg
+	cfg.Token = cloudConfig.Tokens[0]
+	return NewAPIClient(&cfg, c.userAgent), nil
```

Comment:
Looking at
```go
if c.client != nil {
	return c.client, nil
}
```
in the beginning, shouldn't we store this created client in `c.client`? Or don't we to support changing tokens without a restart?

Reply:
alright, thx